src/compiler/base/nodes/tags/content.ts

```diff
@@ -54,13 +53,16 @@ export async function compile(
       ]
   }
 
+  const stray = popStrayText()
+
   if (type === ContentType.text) {
     addContent({
       node,
       content: {
         ...attributes,
         type: ContentType.text,
-        text: textContent,
+        text: stray.text,
+        _promptlSourceMap: stray.sourceMap,
       },
     })
```

Comment:
Add the `includeSourceMap` as a prop in the `CompileNodeContext`, and include it conditionally like you're doing in `compile`.

Reply:
https://github.com/latitude-dev/promptl/blob/c72c963a84b30a0e378ad637b0e1cc9df409686f/src/compiler/base/types.ts#L46